### ISSUE TITLE
NoneType found

### ISSUE THREAD

**Issue by duettra** [h4l/zwiftpower-profile-data#1]
Hi, thanks for the scripts!

All works well for me to the point of executing parse_profile script.

Error: unable to parse profile data from profiles/profile_1332216.html: Unable to parse profiles/profile_1332216.html: sequence item 0: expected str instance, NoneType found
Error: unable to parse profile data from profiles/profile_1707018.html: Unable to parse profiles/profile_1707018.html: sequence item 0: expected str instance, NoneType found

Trying to understand the parse_profiles.py script. Seems it can happen if your list has other stuff in it, but still contains a None. Can you guide me further, please?

If you need more detail in order to help me please ask. 

Thanks a mil!

**Comment by h4l** [comment]
Hey, no worries! I just had a look at this and the problem is that ZwiftPower has updated their profile page HTML structure since I wrote this. It relies on scraping data from the HTML, so it's inevitably pretty brittle code.

I've pushed an update with handles these changes, see: https://github.com/h4l/zwiftpower-profile-data/compare/zp-html-changes

However, the way aliases are listed has changed too (many have no height value), which has downstream implications for the subsequent steps which analyse height deltas. I don't have time to update the rest to work with for these changes, so I'll have to leave that for you to consider. Feel free to fork this repo and investigate that if you're interested.

**Comment by h4l** [comment]
Also the rider list page and rider list API are not working for me since Zwift took ZwiftPower down for maintenance the other day, are they working for you? 

https://zwiftpower.com/api3.php?do=rider_list

Currently the first step doesn't work due to this.

**Comment by duettra** [comment]
Works like a charm, thanks! I dumped the rider list recently to a local file and adjusted the shell scripts to feed them directly. 75MB, 225,139 rider profiles meanwhile.. 

I'm pulling the needed profile data for my TTT team based on individual ZPIDs, so no need for this step anyway.

Height data is available as part of past events. The column might be hidden, but data is in the source file. So far I've gathered it manually from the height column for our small team. If I have any tangible code to contribute, I shall do. Cheers!
